Tests/WorkspaceTests/WorkspaceTests.swift

```diff
@@ -7471,6 +7471,7 @@ final class WorkspaceTests: XCTestCase {
         // make sure artifact downloaded is deleted
         XCTAssertTrue(fs.isDirectory(AbsolutePath("/tmp/ws/.build/artifacts/root")))
         XCTAssertFalse(fs.exists(AbsolutePath("/tmp/ws/.build/artifacts/root/a.zip")))
+        XCTAssertFalse(fs.exists(AbsolutePath("/home/user/caches/org.swift.swiftpm/artifacts/https___a_com_a_zip")))
```

Comment:
Is this the actual path where caches were previously stored? Can we compute it somehow from previously defined paths instead of hardcoding here?

Reply:
My concern here is that if cache storage somehow changes, this test won't detect that and will continue passing.

Reply:
I've updated the test to get the url from the location of the workspace which is also used when initialising the BinaryArtifactsManager in the production code.
The cacheKey is now created the same way as in the BinaryArtifactsManager.

The test could still lead to false positives, if the underlying logic is changed. But I guess to prevent this the BinaryArtifactsManager would need to be refactored which I don't feel comfortable doing at the moment ^^.